qa/rpc-tests/merkle_blocks.py

```diff
@@ -92,6 +92,7 @@ def run_test(self):
         coinbase_txid = result["tx"][0]["txid"]
         result = self.nodes[0].getblock(blockhash, 1)
         assert_equal(coinbase_txid, result["tx"][0])  # verbosity 1 only lists txids
+        assert_equal(result["treestate"], "000000")
```

Comment:
I will like to see some additional test that shows something different than 0.

Reply:
Before creating this PR, I tried to do exactly what you're suggesting, but was unsuccessful. My impression is that this value is zero on regtest, because I added an assertion where this value is retrieved in `rpc/blockchain.cpp` that the value (after serializing) is equal to "000000", ran the entire python test suite, and the assertion never triggered.

Your comment encouraged me to investigate further, so I set and reached a breakpoint at this point in the code (while running a functional test that does a `getblock` rpc), and stepped into `pcoinsTip->GetSaplingAnchorAt()`, and execution reaches `src/coins.cpp:161`, at which time:
```
(gdb) p it->second
$6 = {
  entered = true, 
  tree = {
    static emptyroots = <optimized out>, 
    left = {
      <boost::optional_detail::optional_base<libzcash::PedersenHash>> = {
        <boost::optional_detail::optional_tag> = {<No data fields>}, 
        members of boost::optional_detail::optional_base<libzcash::PedersenHash>: 
        m_initialized = false, 
        m_storage = {
          dummy_ = {
            data = '\000' <repeats 31 times>, 
            aligner_ = {<No data fields>}
          }
        }
      }, <No data fields>}, 
    right = {
      <boost::optional_detail::optional_base<libzcash::PedersenHash>> = {
        <boost::optional_detail::optional_tag> = {<No data fields>}, 
        members of boost::optional_detail::optional_base<libzcash::PedersenHash>: 
        m_initialized = false, 
        m_storage = {
          dummy_ = {
            data = '\000' <repeats 31 times>, 
            aligner_ = {<No data fields>}
          }
        }
      }, <No data fields>}, 
    parents = std::vector of length 0, capacity 0
  }, 
  flags = 0 '\000'
}
```
I must confess to not understanding how to interpret this data structure, or why this value would be zero in regtest.

Another thought I had is that even if the tree state were nonzero, the value may be different for every test run (if some randomness is introduced in the generation of this value), so that would be difficult to verify in a test anyway.

Please let me know if you're aware of a way to improve the testing here, thanks.

Reply:
I will take a more detailed look this weekend and see if i can bring anything that can help.

Reply:
`getblock` is supposed to query any block including the genesis.

The `getblock` from the test is at block 106 where Sapling is not yet activated so it will be always be 0. You can use `GetSproutAnchor()` to get the sporut tree in lower blocks(I made some quick tests and got a sprout tree at block 500 of the mainnet) however still `00000` at block 106(probably for some initial period there was no sprout tree either, need to check the protocol).

Hope it helps.